src/python/gem5/components/boards/kernel_disk_workload.py

```diff
@@ -240,9 +248,13 @@ def set_kernel_disk_workload(
         # Simulator module to setup checkpoints.
         if checkpoint:
             if isinstance(checkpoint, Path):
-                self._checkpoint = checkpoint
-            elif isinstance(checkpoint, CheckpointResource):
-                self._checkpoint = Path(checkpoint.get_local_path())
+                self._checkpoint: Path | None = checkpoint
+            elif (
+                isinstance(checkpoint, CheckpointResource)
+                and (checkpoint_path := checkpoint.get_local_path())
+                is not None
+            ):
+                self._checkpoint = Path(checkpoint_path)
```

Comment:
there are also cases like this where the walrus operator provides a lot more convenience